docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -20,196 +45,193 @@ The scenarios of defining placement rules in SQL include:
+#### Placement Policy
 
-Placement rules must conform to Raft constraints. For example, an error should be reported when executing this statement:
+Creating a new `PLACEMENT POLICY`:
 
 ```sql
-ALTER TABLE test
-	ALTER PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=leader REPLICAS=2;
+CREATE PLACEMENT POLICY `companystandardpolicy` CONSTRAINTS="[+zone=sh]" ROLE=leader REPLICAS=1, CONSTRAINTS="[+zone=sh]" ROLE=follower REPLICAS=1, CONSTRAINTS="[+zone=gz]" ROLE=follower REPLICAS=1;
 ```
 
-There can only be one leader, so `REPLICAS` must be 1 or omitted. But for other roles, `REPLICAS` must be specified.
-
-Besides, at most one role can be defined on the same object. If multiple rules are added on the same role, they will be combined to one rule. For example:
+Creating a new table with a `PLACEMENT POLICY`:
 
 ```sql
-ALTER TABLE test
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=voter REPLICAS=2,
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=bj]" ROLE=voter REPLICAS=2;
+CREATE TABLE t1 (
+	id INT NOT NULL PRIMARY KEY,
+	b VARCHAR(100)
+) PLACEMENT POLICY=`companystandardpolicy`;
 ```
 
-The same role `voter` is defined in 2 different rules, each of which adds 2 replicas. So it is equivalent to:
+Adding `PLACEMENT POLICY` to an existing table:
 
 ```sql
-ALTER TABLE test
-	ADD PLACEMENT POLICY CONSTRAINTS="{+zone=sh:2,+zone=bj:2}" ROLE=voter REPLICAS=4;
+CREATE TABLE t1 (
+	id INT NOT NULL PRIMARY KEY,
+	b VARCHAR(100)
+);
+ALTER TABLE t1 PLACEMENT POLICY=`companystandardpolicy`;
 ```
 
-Note that as there may already exist 3 replicas by default, so it will be 7 replicas after executing this statement. So `ADD PLACEMENT POLICY` can be taken as a shortcut for adding replicas to a defined role. In the example above, it can be replaced by `ALTER PLACEMENT POLICY`.
+Expected Behaviors:
+- `CREATE` or `ALTER` and specifying a `PLACEMENT POLICY` that does not exist results in an error: placement policy 'x' is not defined (TBD: see "unresolved questions")
+- Placement policies are globally unique names. Thus, a policy named `companyplacementpolicy` can apply to the db `test` as well as `userdb`. The namespace does not overlap with other DB objects.
+- Placement Policy names are case insensitive, and follow the same rules as tables/other identifiers for length (64 chars) and special characters.
+- The full placement policy can be seen with `SHOW CREATE PLACEMENT POLICY x`. This is useful for shorthand usage by DBAs, and consistent with other database objects.
+- It is possible to update the definition of a placement policy with `ALTER PLACEMENT POLICY x CONSTRAINTS ..;` This is modeled on the statement `ALTER VIEW` (where the view needs to be redefined)
+- The statement `DROP PLACEMENT POLICY` should return an error if the policy is in use by any objects (TBD: see "unresolved questions")
```

Comment:
Any example for statement `DROP PLACEMENT POLICY` ?

Reply:
Can we add a show policy usage SQL?

e.g. `SHOW PLACEMENT POLICY USAGE FOR x` and list the table, partition, database that used this policy?

Reply:
I have added `SHOW PLACEMENT LIKE 'standardpol%';`, thanks!